main.go

```diff
@@ -49,45 +62,62 @@ var rootCmd = &cobra.Command{
 		for _, architecture := range options.Architectures {
 			content, err := os.Open(options.InputFile)
 			if err != nil {
-				panic(err)
+				return err
 			}
 
 			defer content.Close()
 			result, err := parser.Parse(content)
 			if err != nil {
-				panic(err)
+				return err
 			}
 
 			node := result.AST
 			printNode(node)
 
-			parseNode(node, architecture)
+			color.Cyan("Locking to architecture: %s\n", architecture)
+			err = parseNode(node, architecture)
+			if err != nil {
+				return err
+			}
 
 			var builder strings.Builder
 			writeDockerfile(&builder, node, true)
 			outputName := options.OutputFile
 			if appendArch {
 				outputName = fmt.Sprintf("%s.%s", outputName, architecture)
 			}
-      log.Printf("Writing to %s", outputName)
-			err = os.WriteFile(outputName, []byte(builder.String()), 0600)
-			if err != nil {
-				panic(err)
+			if writeOutput {
+				absPath, err := filepath.Abs(outputName)
+				if err != nil {
+					return err
+				}
+				err = os.WriteFile(outputName, []byte(builder.String()), 0600)
+				if err != nil {
+					return err
+				}
+				color.Green("Generated pinned Dockerfile: %s", absPath)
+			} else {
+				color.Green("Generated pinned Dockerfile\n")
+				fmt.Println(builder.String())
 			}
 		}
+		return nil
 	},
 }
 
 func main() {
 	rootCmd.PersistentFlags().
-		StringP("input", "i", "Dockerfile.template", "Dockerfile to lock down")
+		StringP("input", "i", "Dockerfile.template", "Dockerfile to lock")
 	rootCmd.PersistentFlags().
 		StringP("output", "o", "Dockerfile", "Name of the output dockerfile. If using multiple architectures, the architecture will be appended to the output file name.")
 	rootCmd.PersistentFlags().
-		StringP("architectures", "a", "arm64", "Comma separated list of architectures to lock down to")
+		StringP("architectures", "a", "arm64", "Comma delimited list of architectures to lock")
+	rootCmd.Flags().
```

Comment:
Do we want to make it so that it just prints to the terminal by default? I think we could have a `--dry-run` flag that enables this behaviour and just write the file by default.

Reply:
IMO having it default to stdout seems more safe, and inline with tools we've used before. Since writing overwrites an existing `Dockerfile`, we can consider it a destructive operation. An example worst case is someone tries it out in their repo without any flags and it overwrites a file they didn't intend to.

Reply:
Maybe instead we can add a check to confirm they want to overwrite the dockerfile if it already exists?

Reply:
hmm yeah we can do that. I think if we do that we'd also have to have a `-y` flag to allow it to overwrite non-interactively

Reply:
Yeah exactly.